// @ngx-progressbar/http-client/esm5/ngx-progressbar-http-client.js
import { Injectable, NgModule } from '@angular/core';
import { HTTP_INTERCEPTORS } from '@angular/common/http';
import { finalize } from 'rxjs/operators';
import { NgProgress } from '@ngx-progressbar/core';
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
var NgProgressInterceptor = (function () {
    /**
     * @param {?} progress
     */
    function NgProgressInterceptor(progress) {
        this.progress = progress;
        this.inProgressCount = 0;
    }
    /**
     * @param {?} req
     * @param {?} next
     * @return {?}
     */
    NgProgressInterceptor.prototype.intercept = function (req, next) {
        var _this = this;
        this.inProgressCount++;
        if (!this.progress.isStarted) {
            this.progress.start();
        }
        return next.handle(req).pipe(finalize(function () {
            _this.inProgressCount--;
            if (_this.inProgressCount === 0) {
                _this.progress.done();
            }
        }));
    };
    return NgProgressInterceptor;
}());
NgProgressInterceptor.decorators = [
    { type: Injectable },
];
/** @nocollapse */
NgProgressInterceptor.ctorParameters = function () { return [
    { type: NgProgress, },
]; };
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
var NgProgressHttpClientModule = (function () {
    function NgProgressHttpClientModule() {
    }
    return NgProgressHttpClientModule;
}());
NgProgressHttpClientModule.decorators = [
    { type: NgModule, args: [{
                providers: [
                    { provide: HTTP_INTERCEPTORS, useClass: NgProgressInterceptor, multi: true }
                ],
            },] },
];
/** @nocollapse */
NgProgressHttpClientModule.ctorParameters = function () { return []; };
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
/**
 * Generated bundle index. Do not edit.
 */
export { NgProgressHttpClientModule, NgProgressInterceptor as ɵa };
